### PR TITLE
test,cmake: turn unit.h into unit.cc to speed up compilation

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@cbodley could you please take a look?

File: src/test/osd/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +76,10 @@ target_link_libraries(unittest_hitset osd global ${BLKID_LIBRARIES})
 add_executable(unittest_osd_osdcap
   osdcap.cc
 )
+if(HAS_VTA)
+  set_source_files_properties(osdcap.cc PROPERTIES
+    COMPILE_FLAGS -fno-var-tracking-assignments)
+endif()
```
Comment (cbodley) [inline src/test/osd/CMakeLists.txt]: this looks unrelated

Comment (tchaikov) [inline src/test/osd/CMakeLists.txt]: @cbodley i don't understand why this is unrelated. as the commit message suggests, by disabling `var-tracking-assignments`, we can disable the warning and avoid recompiling this file.

Comment (cbodley) [inline src/test/osd/CMakeLists.txt]: ok thanks, i didn't see the commit message. was that warning already present, or somehow introduced by moving unit.h -> unit.cc?

Comment (tchaikov) [inline src/test/osd/CMakeLists.txt]: that warning was already present, see 
```
[ 74%] Building CXX object src/tools/cephfs/CMakeFiles/cephfs-table-tool.dir/RoleSelector.cc.o
In file included from /srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/googletest/googletest/include/gtest/gtest.h:58:0,
from /srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/test/osd/osdcap.cc:20:
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/test/osd/osdcap.cc: In member function 'virtual void OSDCap_AllowClassMulti_Test::TestBody()':
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/test/osd/osdcap.cc:766:6: note: variable tracking size limit exceeded with -fvar-tracking-assignments, retrying without
TEST(OSDCap, AllowClassMulti) {
^
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/googletest/googletest/include/gtest/internal/gtest-internal.h:1211:3: note: in definition of macro 'GTEST_TEST_CLASS_NAME_'
test_case_name##_##test_name##_Test
^
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/googletest/googletest/include/gtest/gtest.h:2181:3: note: in expansion of macro 'GTEST_TEST_'
GTEST_TEST_(test_case_name, test_name, \
^
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/googletest/googletest/include/gtest/gtest.h:2187:42: note: in expansion of macro 'GTEST_TEST'
# define TEST(test_case_name, test_name) GTEST_TEST(test_case_name, test_name)
^
/srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILD/ceph-11.0.2/src/test/osd/osdcap.cc:766:1: note: in expansion of macro 'TEST'
TEST(OSDCap, AllowClassMulti) {
^
Linking CXX executable ../../../bin/unittest_osd_osdcap
```

at http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-rpm-centos7-amd64-notcmalloc/log.cgi?log=64dd4fe92a3955fe7d98d3de77f7c5174893c745
